moz_security/sandboxes/heka/analysis/moz_security_cloudtrail.lua

```diff
@@ -0,0 +1,147 @@
+message_matcher =
+ticker_interval = 0
+process_message_inject_limit = 1
+
+-- cloudtrail events to alert on (required)
+events = {
+    {
+        description = "mfa disabled",
+        fields = {
+            { "eventName", "^DeleteVirtualMFADevice$" },
+        }
+    },
+    {
+        description = "mfa disabled",
+        fields = {
+            { "eventName", "^DeactivateMFADevice$" },
+        }
+    },
+    {
+        description = "access key created",
+        fields = {
+            { "eventName", "^CreateAccessKey$" }
+        }
+    },
+    {
+        description = "IAM action in production account from console without mfa",
+        fields = {
+            { "eventSource", "^iam.amazonaws.com$" },
+            { "recipientAccountId", "^1234567890$" },
+            { "userIdentity.invokedBy", "^signin.amazonaws.com$" },
+            { "userIdentity.sessionContext.attributes.mfaAuthenticated", "^false$" }
+        }
+    }
+}
+
+-- mapping of aws account ids to human-friendly names (optional)
+aws_account_mapping = {
+    {"5555555555", "dev"},
+    {"1234567890", "prod"}
+}
+
+
+-- module makes use of alert output and needs a valid alert configuration
+alert = {
+    modules = { }
+}
+```
+--]]
+
+require "cjson"
+
+local alert = require "heka.alert"
+
+local cfevents = read_config("events")
+local cfaccount_name_mapping = read_config("aws_account_mapping")
+
+local payloadfields = {
+    "awsRegion",
+    "eventID",
+    "eventName",
+    "eventSource",
+    "eventTime",
+    "eventType",
+    "eventVersion",
+    "recipientAccountId",
+    "requestID",
+    "requestParameters.resourceName",
+    "requestParameters.userName",
+    "responseElements.accessKey.accessKeyId",
+    "responseElements.accessKey.createDate",
+    "responseElements.accessKey.status",
+    "responseElements.accessKey.userName",
+    "sourceIPAddress",
+    "userAgent",
+    "userIdentity.accessKeyId",
+    "userIdentity.accountId",
+    "userIdentity.arn",
+    "userIdentity.invokedBy",
+    "userIdentity.principalId",
+    "userIdentity.sessionContext.attributes.creationDate",
+    "userIdentity.sessionContext.attributes.mfaAuthenticated",
+    "userIdentity.sessionContext.sessionIssuer.accountId",
+    "userIdentity.sessionContext.sessionIssuer.arn",
+    "userIdentity.sessionContext.sessionIssuer.principalId",
+    "userIdentity.sessionContext.sessionIssuer.type",
+    "userIdentity.sessionContext.sessionIssuer.userName",
+    "userIdentity.type",
+    "userIdentity.userName"
+}
+
+function genpayload()
+    local det = {}
+    for _,v in ipairs(payloadfields) do
+        local vk = string.format("Fields[%s]", v)
+        det[v] = read_message(vk)
+    end
+    return cjson.encode(det)
+end
+
+function get_account_name(account_id)
+    -- { {account_id, account_name}, ... }
+    if cfaccount_name_mapping ~= nil then
+      for _, v in ipairs(cfaccount_name_mapping) do
+          if v[1] == account_id then return v[2] end
```

Comment:
if we change cfaccount_name_mapping to be a hash we can just `return cf_account_name_mapping[k] or account_id` and skip the loop

Reply:
I wanted to do this, but I was getting parsing errors when there was a number as the key.

both
```
aws_account_mapping = {
    5555555555 = "dev",
    1234567890 = "prod"
}
```
and
```
aws_account_mapping = {
    "5555555555" = "dev",
    "1234567890" = "prod"
}
```
were throwing configuration parsing errors.

Is there documentation on the config file format or is it just a lua file?

Reply:
I think you can specify like:

```
aws_account_mapping = {
    ["5555555555"] = "dev"
}
```

It's just a lua file